pkg/api/mlflow/dao/repositories/run.go

```diff
@@ -89,13 +93,43 @@ func (r RunRepository) GetByIDAndLifecycleStage(
 		"Params",
 	).Preload(
 		"Tags",
+	).Joins(
+		"LEFT JOIN experiments ON experiments.experiment_id = runs.experiment_id",
+	).Where(
+		"experiments.namespace_id = ?", namespaceID,
 	).Where(
-		`lifecycle_stage = ?`, lifecycleStage,
+		`runs.lifecycle_stage = ?`, lifecycleStage,
 	).First(&run).Error; err != nil {
 		if errors.Is(err, gorm.ErrRecordNotFound) {
 			return nil, nil
 		}
-		return nil, eris.Wrapf(err, "error getting 'run' entity by id: %s", id)
+		return nil, eris.Wrapf(err, "error getting 'run' entity by id: %s", runID)
+	}
+	return &run, nil
+}
+
+// GetByNamespaceIDAndRunID returns models.Run entity by Namespace ID and its ID.
+func (r RunRepository) GetByNamespaceIDAndRunID(
```

Comment:
Testme.

Reply:
no, we usually don't need to test repositories. integration tests do this for us.